.github/workflows/deploy_branches.yml

```diff
@@ -1,10 +1,49 @@
 # General notes on github actions: Note that both the working directory
 # and environment variables generally are not shared between steps.
-name: Deploy Hazel
+name: Build and Deploy Hazel
 on: [push]
 jobs:
+  Build:
+    runs-on: ubuntu-latest
+    steps:
+      # NOTE: position the below lines in the code between two steps
+      # and uncomment them to open an ssh connection at that point:
+      #- name: Debugging with ssh
+      #  uses: lhotari/action-upterm@v1
+      - name: Checkout the hazel repo on the current branch # STEP 1
+        uses: actions/checkout@v2
+        with:
+          path: source
+      - name: Add the name of the current branch to the environment as BRANCH_NAME
+        uses: nelonoel/branch-name@v1.0.1
+      - name: Retrieve the build environment if cached # STEP 2
+        id: opam-cache
+        uses: actions/cache@v2
+        with:
+          path: '/home/runner/.opam/'
+          key: ${{ runner.os }}-modules-${{ hashFiles('./source/opam.export') }}
+      - name: Install dependencies
+        run: |
+          sudo apt --assume-yes install curl m4 opam
+          export OPAMYES=1
+          opam init --compiler=ocaml-base-compiler.5.0.0
+          eval $(opam env)
+          make deps 
+        working-directory: ./source
+      - name: Build Hazel
+        run: |
+          eval $(opam env)
+          make release
+        working-directory: ./source
+      - name: Cache the build
+        id: build-cache
+        uses: actions/cache@v4
+        with:
+          path: ./source/_build'
+          key: ${{ runner.os }}-build-${{ hashFiles('./source/**', '!./source/_build/**', '!./source/.git/**') }}
```

Comment:
This isn't working. I'm guessing `_build` is cached using file timestamps instead of a hash of the source files so it rebuilds everything. Going to try https://github.com/marketplace/actions/set-up-ocaml with the dune-cache